RUFAS/routines/animal/ration/ration_driver.py

```diff
@@ -145,13 +159,22 @@ def formulate_ration(
                     dict(info_map, **{"units": fail_summary_units}),
                 )
 
-        if solution is not None:
+        if solution.success and solution is not None:
```

Comment:
```suggestion
        if solution is not None and solution.success:
```

To avoid a potential error of checking for an attribute first if solution is None.